src/config/unit_test.rs

```diff
@@ -503,37 +503,15 @@ async fn build_unit_test(
                 .iter()
                 .enumerate()
             {
-                match cond_conf {
-                    TestCondition::Embedded(b) => match b.build() {
-                        Ok(c) => {
-                            conditions.push(c);
-                        }
-                        Err(e) => {
-                            errors.push(format!(
-                                "failed to create test condition '{}': {}",
-                                index, e,
-                            ));
-                        }
-                    },
-                    TestCondition::NoTypeEmbedded(n) => match n.build() {
-                        Ok(c) => {
-                            conditions.push(c);
-                        }
-                        Err(e) => {
-                            errors.push(format!(
-                                "failed to create test condition '{}': {}",
-                                index, e,
-                            ));
-                        }
-                    },
-                    TestCondition::String(_s) => {
-                        errors.push(format!(
-                            "failed to create test condition '{}': condition references are not yet supported",
-                            index
-                        ));
-                    }
+                match cond_conf.build() {
+                    Ok(c) => conditions.push(c),
+                    Err(e) => errors.push(format!(
+                        "failed to create test condition '{}': {}",
```

Comment:
Do we have a guideline on how errors should be formatted?

Not sure if directly relatied, but the [logging-style](https://github.com/timberio/vector/blob/master/CONTRIBUTING.md#logging-style) says to capitalize and end with a period.

Reply:
I haven't changed the error message in this PR, just moved it to a different place so I am hesitant to change it here. I think because this message is part of a larger message that will ultimately be reported to the user, that is why it hasn't been capitalized here..